proposals/0721-golden-template-crnl.md

```diff
@@ -0,0 +1,178 @@
+
+1. **Updating `create-react-native-library`:** The first step is to update the **`create-react-native-library`** tool to make the official template the recommended option.
+2. **Communicating with library owners:** To encourage existing React Native libraries to adopt the official template, proactive communication is essential. Library owners can be informed about the advantages of the official template, such as easier upgrades and better compatibility with future React Native versions. The communication might also include step-by-step guides on migrating existing libraries to the official template and utilizing the library upgrade helper for seamless upgrades. Further details won’t be covered here.
+3. **Curating example libraries**: To promote adoption, we can compile a list of outstanding libraries that we intend to target for adopting the official template. These libraries will also serve as practical examples of best practices in React Native development, demonstrating the template's effectiveness across various use cases. The `create-react-native-library team` can proactively engage with these libraries to ensure compatibility. An early example of such a library could be [React Native Gesture Handler](https://github.com/software-mansion/react-native-gesture-handler/).
+
+## **Documentation update is needed**
+
+Updating the React Native documentation is crucial to ensure that developers have comprehensive and up-to-date information about the official library template and the library upgrade helper. The following documentation updates are required:
+
+1. **Official Library Template:** The documentation should include detailed instructions on how to use the **`create-react-native-library`** CLI tool with the official library template. Additionally, developers should be provided with examples and best practices to help them get started quickly.
+2. **Library Upgrade Helper:** A dedicated section can be added to the documentation to explain the library upgrade helper's functionality. It should describe how the helper compares different library versions, provides upgrade instructions, and assists in ensuring compatibility with newer React Native releases. The documentation might offer step-by-step guides on how to utilize the helper effectively.
+3. **Benefits and Advantages:** The documentation should clearly outline the benefits of adopting the official template and using the library upgrade helper. This section should emphasize the importance of keeping libraries up-to-date with the latest React Native versions, the improved stability resulting from using the official template, and how the helper streamlines the upgrade process.
+
+## **Unresolved questions**
+
+- How to encourage existing React Native libraries to use the official template?
+- How should we tackle the support for out-of-tree platforms such as [React Native Windows](https://github.com/microsoft/react-native-windows)?
```

Comment:
good question, I've relayed the RFC internally so that folks from that team can hop in and give their thoughts.

Reply:
nothing has been said, I can ask flag it up once again but I wouldn't consider it a blocker